ui/selectors/networks.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  RpcEndpointType,
+  type NetworkConfiguration,
+  type NetworkState as _NetworkState,
+} from '@metamask/network-controller';
+import { createSelector } from 'reselect';
+import { NetworkStatus } from '../../shared/constants/network';
+import { createDeepEqualSelector } from './util';
+
+export type NetworkState = { metamask: _NetworkState };
+
+export type NetworkConfigurationsState = {
+  metamask: {
+    networkConfigurations: Record<
+      string,
+      MetaMaskExtensionNetworkConfiguration
+    >;
+  };
+};
+
+export type SelectedNetworkClientIdState = {
+  metamask: {
+    selectedNetworkClientId: string;
+  };
+};
+
+export type MetaMaskExtensionNetworkConfiguration = NetworkConfiguration;
+
+export type NetworkConfigurationsByChainIdState = {
+  metamask: Pick<_NetworkState, 'networkConfigurationsByChainId'>;
+};
+
+export type ProviderConfigState = NetworkConfigurationsByChainIdState &
+  SelectedNetworkClientIdState;
+
+export const getNetworkConfigurationsByChainId = createDeepEqualSelector(
+  (state: NetworkConfigurationsByChainIdState) =>
+    state.metamask.networkConfigurationsByChainId,
+  (networkConfigurationsByChainId) => networkConfigurationsByChainId,
+);
```

Comment:
We decided that as we refactor, anything that is a selector should start with `select` to better follow the guidelines.